drgn_tools/task.py

```diff
@@ -234,4 +351,4 @@ def run(self, prog: Program, args: argparse.Namespace) -> None:
         if args.last_run:
             show_tasks_last_runtime(for_each_task(prog))
         else:
-            raise NotImplementedError("currently, only ps -m is implemented")
+            show_taskinfo(prog, for_each_task(prog))
```

Comment:
So, this isn't quite everything in terms of the `ps` command we use in crash. And don't get me wrong, I don't think it needs to be identical, or contain every option. But there are some core things that would be nice. In particular:

* allow passing `pid | task | command` to filter by these. So you could do `ps java` to see all java process, or `ps 1` to see all threads of init, etc.
* `-k` filter to only kernel threads, or `-u` filter to only user threads
* `-G` display only the thread group leader (i.e task where `pid == ppid`)

The nice thing is that these can all be implemented in a straightforward way that allows everything to combine nicely:

```
tasks = for_each_task(prog)

if args.user:
    tasks = filter(tasks, is_user)
elif args.kernel:
    tasks = filter(tasks, is_kthread)
elif args.group_leader:
    tasks = filter(tasks, is_group_leader)

if args.last_run:
    show_tasks_last_runtime(tasks)
else:
    show_taskinfo(prog, tasks)
```

And this could also be extended to support some of the other ps options, if you wanted.